src/option_parser.c

```diff
@@ -170,36 +170,77 @@ int string_parse_sepcolor(const void *data, const char *s, void *ret)
 {
         LOG_D("parsing sep_color");
         struct separator_color_data *sep_color = (struct separator_color_data*) ret;
+        struct color invalid = COLOR_UNINIT;
 
         enum separator_color type;
         bool is_enum = string_parse_enum(data, s, &type);
         if (is_enum) {
                 sep_color->type = type;
-                g_free(sep_color->sep_color);
-                sep_color->sep_color = NULL;
+                sep_color->color = invalid;
                 return true;
         } else {
                 if (STR_EMPTY(s)) {
                         LOG_W("Sep color is empty, make sure to quote the value if it's a color.");
                         return false;
                 }
-                if (s[0] != '#') {
-                        LOG_W("Sep color should start with a '#'");
-                        return false;
-                }
-                if (strlen(s) < 4) {
-                        LOG_W("Make sure the sep color is formatted correctly");
-                        return false;
-                }
-                // TODO add more checks for if the color is valid
 
-                sep_color->type = SEP_CUSTOM;
-                g_free(sep_color->sep_color);
-                sep_color->sep_color = g_strdup(s);
-                return true;
+                if (string_parse_color(NULL, s, &sep_color->color)) {
+                        sep_color->type = SEP_CUSTOM;
+                        return true;
+                }
         }
+        return false;
 }
 
+#define UINT_MAX_N(bits) ((1 << bits) - 1)
+
+// Parse a #RRGGBB[AA] string
+int string_parse_color(const void *data, const char *s, void *ret)
```

Comment:
```suggestion
int string_parse_color(const void *data, const char *s, struct color *color)
```

Shouldn't the target always be a `struct color *`? Then we should be able to get rid of line 205 as well.

Reply:
Yes, I was trying to use TYPE_CUSTOM before but then I changed